### ISSUE TITLE
Unify fix logic for position of `ON` keyword

### ISSUE THREAD

**Issue by tkachenkomaria244** [sqlfluff/sqlfluff#4063]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Need to have possibility to unify fixing position of `ON` keyword. 
We expect newline after ON even if there's only one join condition.
Here is code example:
```
select a.fld5
from source_data as a
left join source_data as b on
	a.fld1 = b.fld1
left join source_data as c
	on a.fld1 = c.fld1
```

### Expected Behaviour

Fixed code we expect:
```
select a.fld5
from source_data as a
left join
	source_data as b
	on
		a.fld1 = b.fld1
left join
	source_data as c
	on
		a.fld1 = c.fld1
```

### Observed Behaviour

No issues are found/fixed:
```
==== finding fixable violations ====
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

### How to reproduce

Run with `sqlfluff fix D:\v_v.sql --rules L001,L002,L003,L004,L005,L006,L008,L009,L036,L050 --config config\sqlfluff\.sqlfluff`

### Dialect

tsql, postgres

### Version

sqlfluff 1.4.1
python 3.9.13

### Configuration

[sqlfluff]
sql_file_exts = .sql
dialect = tsql
templater = placeholder

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

[sqlfluff:templater:placeholder]
param_regex = \$\{(?P<param_name>[\w_]+)\}
V_LOAD_ID = '1111'
V_LOAD_DATE_END = "2020-01-29 08:52:11.0698120"
V_LOAD_DATE_START = "2020-01-29 08:52:11.0698120"
V_TARGET_LAND_SCHEMA = dl
V_TARGET_LAND_TABLE = tbl1
V_LOAD_TYPE = delta
V_REF_LOAD_ID_MIN = '1000'
V_REF_LOAD_ID_MAX = '2000'
V_TARGET_SCHEMA = schema1
V_TARGET_TABLE = table1
PART_SIZE = '10'
SEQUENCE_NUM = '1'

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 250
indent_unit = tab
single_table_references = qualified
allow_scalar = False

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:layout:type:keyword]
line_position = alone:strict

[sqlfluff:rules:L010] # Keywords
capitalisation_policy = lower

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L030] # Function names
capitalisation_policy = lower
[sqlfluff:rules:L040] # Null & Boolean Literals
capitalisation_policy = lower

[sqlfluff:rules:L028]
single_table_references = qualified

[sqlfluff:rules:L063] # datatypes
extended_capitalisation_policy = lower

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
